### PR TITLE
Sql injection

### PR DESCRIPTION
**Pull request by MatthewOakley**
Just did some small work the biggest thing stalling me is master code can not compile. The work that was done was to add the logging if a user tried to use an sql attack. The keywords for checking this are very basic right now but will be improved later on. I will also improve the instant and userAgent format to get a more readable output.

### CODE REVIEW

File: src/main/java/codeu/model/store/basic/MessageStore.java [hunk 1]
```diff
@@ -130,14 +130,4 @@ public Message getMessageById(UUID id) {
...
-
-  /** Returns message by Id */
-  public Message getMessageById(UUID id) {
-    for (Message message : messages) {
-      if (message.getId().equals(id)) { 
-        return message; 
-      }
-    }
-    return null; 
-  }
```
Comment (StevenAb) [inline src/main/java/codeu/model/store/basic/MessageStore.java]: Why did you delete this method?

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 2]
```diff
@@ -170,27 +170,28 @@ public PersistentDataStore() {
...
   
-      for (Entity entity : results.asIterable()) {
-          try {
-
-            Set<String> dataStoreMessageIds = new HashSet<>((Collection<String>)
+    PreparedQuery results = datastore.prepare(query);
+  
+    for (Entity entity : results.asIterable()) {
+      try {
+        Set<String> dataStoreMessageIds = new HashSet<>((Collection<String>)
```
Comment (StevenAb) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java:190]: Is the datastore reading/writing the Sets? I was also having compilation problems and it was related to trying to read/write other Collections like ArrayList

**Review by StevenAb — COMMENTED** [review]
Had a couple questions, but most of the code LGTM!

File: src/main/java/codeu/model/store/basic/MessageStore.java (hunk 1 above)
Comment (MatthewOakley) [reply src/main/java/codeu/model/store/basic/MessageStore.java]: We had two of the same methods in this file

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java (hunk 2 above)
Comment (MatthewOakley) [reply src/main/java/codeu/model/store/persistence/PersistentDataStore.java:190]: It is reading from datastore but the problem previously is that we made a query and never ran it. So it was never getting any results and was not running this for loop For the error you are getting is this when you pulled from master or before?

**Review by StevenAb — APPROVED** [review]

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 3]
```diff
@@ -55,11 +59,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
+            .withLocale(Locale.US)
+            .withZone(ZoneId.systemDefault());
+        Instant instant = Instant.now();
+        String time = formatter.format(instant);
+        writer.append(time + "\n");
+      } catch (Exception e) {
+        e.printStackTrace();
+      } finally {
+        writer.close();
+
```
Comment (ilianac) [inline src/main/java/codeu/controller/RegisterServlet.java]: delete

**Review by ilianac — APPROVED** [review]
Amazing! Looks good! Just one tiny space that could be deleted.

**Comment by nanselmo** [comment]
This looks good. Thanks to you all for reviewing so quickly. If you have time, I would think about packaging the isAttack into it's own little function. Right now it looks like you're just checking the username and not the password. If you make this it's own little function, then you can check any parameter being sent in a POST request - including those from other views. That said, plan on deploying this tomorrow and you can always open a new PR to add additional functionality. Also I'm super new to your project/Java/applets, so take that suggestion with more than a few grains of salt. Maybe a cup of salt :)
